bot/logger/logger.py

```diff
@@ -0,0 +1,60 @@
+"""Logger Module. Configures the logger for the discord library and the logger for custom code.
+It also provides the decorator for logging command calls.
+"""
+import functools
+import logging
+
+import discord
+
+from bot import constants
+
+# configure logger for bot
+log = logging.getLogger("bot")
+log.setLevel(logging.INFO)
+file_handler = logging.FileHandler(filename=constants.LOG_FILE_PATH, encoding='utf-8', mode='w')
+file_handler.setFormatter(logging.Formatter('%(levelname)s %(asctime)s: %(message)s'))
+log.addHandler(file_handler)
+stream_handler = logging.StreamHandler()
+stream_handler.setLevel(logging.WARNING)
+stream_handler.setFormatter(logging.Formatter('%(levelname)s %(asctime)s: %(message)s'))
+log.addHandler(stream_handler)
+
+# add discord to logging (file and console)
+discord_logger = logging.getLogger('discord')
+discord_logger.setLevel(logging.WARNING)
+discord_logging_file_handler = logging.FileHandler(filename=constants.LOG_FILE_PATH, encoding='utf-8', mode='w')
+discord_logging_file_handler.setFormatter(logging.Formatter('%(asctime)s:%(levelname)s:%(name)s: %(message)s'))
+discord_logger.addHandler(discord_logging_file_handler)
+discord_logging_stream_handler = logging.StreamHandler()
+discord_logging_stream_handler.setFormatter(logging.Formatter('%(asctime)s:%(levelname)s:%(name)s: %(message)s'))
+discord_logger.addHandler(discord_logging_stream_handler)
+
+
+def command_log(func):
+    """Decorator to log when a command is triggered.
+    If a command function is annotated with '@command_log' it will automatically log if it is called, by whom and in
```

Comment:
Add a blank line here.

Reply:
Why in the middle of a sentence?

Reply:
No. Between the header of the dosctring and the description.

Reply:
Ahh